src/jquery/attributes.js

```diff
@@ -5,10 +5,24 @@ var oldRemoveAttr = jQuery.fn.removeAttr,
 	rmatchNonSpace = /\S+/g;
 
 migratePatchFunc( jQuery.fn, "removeAttr", function( name ) {
-	var self = this;
+	var self = this,
+		patchNeeded = false;
 
 	jQuery.each( name.match( rmatchNonSpace ), function( _i, attr ) {
 		if ( jQuery.expr.match.bool.test( attr ) ) {
+
+			// Only warn if at least a single node had the property set to
+			// something else than `false`. Otherwise, this Migrate patch
+			// doesn't influence the behavior and there's no need to set or warn.
+			self.each( function() {
+				if ( jQuery( this ).prop( attr ) !== false ) {
```

Comment:
Could we just use `this.attr !== false` here?

Reply:
I was playing safe to not omit any possible prop hooks. We _almost_ don't have any for boolean props... except for `selected` in IE:
https://github.com/jquery/jquery/blob/3.6.3/src/attributes/prop.js#L91-L125
It seems we should use the jQuery API then, shouldn't we?

Reply:
Yeah I suppose we might add a prop hook so this is the safest way. 👍